_episodes/02-image-basics.md

```diff
@@ -97,54 +60,214 @@ color, but that each pixel can have a different color from its neighbors.
+>as they go down instead of up as in a normal Cartesian coordinate system.
+{: .callout }
+> ## Changing Pixel Values (5 min)
+>
+> Load another copy of eight named five, and then 
+> change the value of pixels so you have what looks like a 5 instead of 8. Dislay the image and print out the matrix as well.
+>
+> > ## Solution
+> >There are many possible solutions, but one method would be . . .
+> >~~~
+> >five = skimage.io.imread(fname="data/eight.tif")
+> >five[1,2]= 1.0
+> >five[3,0]= 1.0
+> >fig, ax = plt.subplots()
+> >plt.imshow(five)
+> >print(five)
+> >~~~
+> >{: .language-python}
+> > ![Image of 5 matrix](../fig/02-matrix-5.png)
+> > ![Image of 5](../fig/02-five.png)
+> {: .solution}
+{: .challenge}
 
-Most frequently, the amount of the primary color added is represented as
-an integer in the closed range [0, 255]. Therefore, there are 256 discrete
+## More colors
+Up to now, we only had a 2 color matrix, but we can have more if we use other numbers or fractions. One common way is to use the numbers between 0 and 255 to allow for 256 different colors or 256 different levels of grey.  Let's try that out.
+ ~~~
+#make a copy of eight
+three_colors = skimage.io.imread(fname="data/eight.tif")
+
+#multiply the whole matrix by 128
+three_colors = three_colors * 128
+
+# set the middle row (index 2) to the value of 255., so you end up with the values 0.,128.,and 255
+three_colors[2,:] = 255.
+fig, ax = plt.subplots()
+plt.imshow(three_colors)
+print(three_colors)
+~~~
+{: .language-python}
+![Image of three colors](../fig/02-three-colors.png)
+We now have 3 colors, but are they the three colors you expected?  They all appear to be on a continum of dark purple on the low end and yellow on the high end. This is a consequence of the default color map (cmap) in this library. You can think of a color map as an association or mapping of numbers to a specific color. However, the goal here is not to have one number for every color under the sun, but rather to have a continuum of colors that demonstrate relative intensity. In our specific case here for example, 255 or the highest intensity is mapped to yellow, and 0 or the lowest intensity is mapped to a dark purple. The best color map for your data will vary and there are many options built in, but this default selection was not arbitrary.  A lot of science went into making this the default due to its robustness when it comes to how the human mind interprets relative color values, grey-scale printability, and color-blind friendliness ([https://matplotlib.org/stable/tutorials/colors/colormaps.html](https://matplotlib.org/stable/tutorials/colors/colormaps.html), [https://bids.github.io/colormap/](https://bids.github.io/colormap/)).  Thus it is a good place to start, and you should change it only with purpose and forethought.  For now, let's see how you can do that using an alternative map you have likley seen before where it will be even easier to see it as a mapped continuum of intensities: greyscale.
```

Comment:
```suggestion
We now have 3 colors, but are they the three colors you expected?  They all appear to be on a continuum of dark purple on the low end and yellow on the high end. This is a consequence of the default color map (cmap) in this library. You can think of a color map as an association or mapping of numbers to a specific color. However, the goal here is not to have one number for every possible color, but rather to have a continuum of colors that demonstrate relative intensity. In our specific case here for example, 255 or the highest intensity is mapped to yellow, and 0 or the lowest intensity is mapped to a dark purple. The best color map for your data will vary and there are many options built in, but this default selection was not arbitrary.  A lot of science went into making this the default due to its robustness when it comes to how the human mind interprets relative color values, grey-scale printability, and color-blind friendliness ([https://matplotlib.org/stable/tutorials/colors/colormaps.html](https://matplotlib.org/stable/tutorials/colors/colormaps.html), [https://bids.github.io/colormap/](https://bids.github.io/colormap/)).  Thus it is a good place to start, and you should change it only with purpose and forethought.  For now, let's see how you can do that using an alternative map you have likely seen before where it will be even easier to see it as a mapped continuum of intensities: greyscale.
```

Reply:
Hey, Erin

Thanks for all the good feedback and suggestions.  I am working through them via the github interface, and nearly all them were easy to understand, but a couple of them puzzle me like this one.  Not sure if it is some artifact of the git or if I am just no seeing it.  This looks like no change to my eyes, so what am I missing if anything?